middleware/src/middleware.go

```diff
@@ -164,3 +164,94 @@ func (middleware *Middleware) SampleInfo() rpcmessages.SampleInfoResponse {
 func (middleware *Middleware) VerificationProgress() rpcmessages.VerificationProgressResponse {
 	return middleware.verificationProgress
 }
+
+// Flashdrive returns a FlashdriveResponse struct in response to a rpcserver request
+func (middleware *Middleware) Flashdrive(args rpcmessages.FlashdriveArgs) (rpcmessages.FlashdriveResponse, error) {
+	switch args.Method {
+	case rpcmessages.Check:
+		log.Println("Executing a USB flashdrive check via the cmd script")
+		out, err := middleware.runBBBCmdScript("flashdrive", "check")
+		if err != nil {
+			return rpcmessages.FlashdriveResponse{Success: false, Message: string(out)}, nil
```

Comment:
Why not return the error as well since you already have it as one of the return types?
Same for other calls below. 
@TheCharlatan I need to catch up with you at some point what kind of error handling you decided on :)

Reply:
Good point as well. Makes no sense to return `nil` everywhere.